### PR TITLE
[Flink AI Flow] Add java client for metric service

### PR DESCRIPTION
**Pull request by jiangxin369**
## What is the purpose of the change

Add java client for metric service

## Verifying this change

*(Please pick either of the following options)*

This change added tests.

### CODE REVIEW

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java [hunk 1]
```diff
@@ -926,4 +929,167 @@ public void stopListenEvent(String namespace, String key) {
...
+     * @param datasetName   Name of the dataset associated with the registered metric meta.
+     * @param modelName     Name of the model associated with the registered metric meta.
+     * @param jobName       Name of the job associated with the registered metric meta.
+     * @param startTime     Start time of registered metric meta.
+     * @param endTime       End time of registered metric meta.
+     * @param uri           Uri of registered metric meta.
+     * @param tags          Tags of registered metric meta.
+     * @param properties    Properties of registered metric meta.
+     *
+     * @return A single :py:class:`ai_flow.meta.metric_meta.MetricMeta` object.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java:949]: Update the return type of `registerMetricMeta`.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java [hunk 2]
```diff
@@ -926,4 +929,167 @@ public void stopListenEvent(String namespace, String key) {
...
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java:971]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java [hunk 3]
```diff
@@ -926,4 +929,167 @@ public void stopListenEvent(String namespace, String key) {
...
+     */
+    public boolean deleteMetricMeta(String metricName) {
+        return this.metricClient.deleteMetricMeta(metricName);
+    }
+
+    /***
+     * Get metric metadata detail filter by metric name for Metric Center.
+     *
+     * @param metricName Name of registered metric meta. This is expected to be unique in the backend store.
+     * @return A single :py:class:`ai_flow.meta.metric_meta.MetricMeta` object.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java:993]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java [hunk 4]
```diff
@@ -926,4 +929,167 @@ public void stopListenEvent(String namespace, String key) {
...
+    public MetricMeta getMetricMeta(String metricName) throws AIFlowException {
+        return this.metricClient.getMetricMeta(metricName);
+    }
+
+    /***
+     * List dataset metric metadata filter by dataset name and project name for Metric Center.
+     *
+     * @param datasetName Name of the dataset associated with the registered metric meta.
+     * @param projectName Name of the project associated with the registered metric meta.
+     * @return List of :py:class:`ai_flow.meta.metric_meta.MetricMeta` objects.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java:1005]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java [hunk 5]
```diff
@@ -926,4 +929,167 @@ public void stopListenEvent(String namespace, String key) {
...
+    public List<MetricMeta> listDatasetMetricMetas(String datasetName, String projectName) throws AIFlowException {
+        return this.metricClient.listDatasetMetricMetas(datasetName, projectName);
+    }
+
+    /***
+     * List model metric metadata filter by model name and project name for Metric Center.
+     *
+     * @param modelName     Name of the model associated with the registered metric meta.
+     * @param projectName   Name of the project associated with the registered metric meta.
+     * @return List of :py:class:`ai_flow.meta.metric_meta.MetricMeta` objects.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java:1017]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java [hunk 6]
```diff
@@ -926,4 +929,167 @@ public void stopListenEvent(String namespace, String key) {
...
+    /***
+     * Register metric summary in Metric Center.
+     *
+     * @param metricName        Name of registered metric summary.
+     * @param metricKey         Key of registered metric summary.
+     * @param metricValue       Value of registered metric summary.
+     * @param metricTimestamp   Timestamp of registered metric summary.
+     * @param modelVersion      Version of the model version associated with the registered metric summary.
+     * @param jobExecutionId    ID of the job execution associated with the registered metric summary.
+     * @return A single :py:class:`ai_flow.meta.metric_meta.MetricSummary` object.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java:1033]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java [hunk 7]
```diff
@@ -926,4 +929,167 @@ public void stopListenEvent(String namespace, String key) {
...
+     *
+     * @return A single :py:class:`ai_flow.meta.metric_meta.MetricMeta` object.
+     */
+    public MetricMeta registerMetricMeta(String name, MetricType metricType, String projectName, String description, String datasetName, String modelName,
+                                         String jobName, long startTime, long endTime, String uri, String tags, Map<String, String> properties) throws AIFlowException {
+        return this.metricClient.registerMetricMeta(name, metricType, projectName, description, datasetName, modelName, jobName, startTime, endTime, uri, tags, properties);
+    }
+
+    /**
+     * Update a MetricMeta in metric center.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java:957]:
```suggestion
     * Update a MetricMeta in Metric Center.
```

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java [hunk 8]
```diff
@@ -926,4 +929,167 @@ public void stopListenEvent(String namespace, String key) {
     public long getLatestVersion(String namespace, String key) throws Exception {
         return this.notificationClient.getLatestVersion(namespace, key);
     }
+
+    /**
+     * Register a MetricMeta in metric center.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java:934]:
```suggestion
     * Register a MetricMeta in Metric Center.
```

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java [hunk 9]
```diff
@@ -926,4 +929,167 @@ public void stopListenEvent(String namespace, String key) {
...
+     */
+    public boolean deleteMetricSummary(long uuid) {
+        return this.metricClient.deleteMetricSummary(uuid);
+    }
+
+    /***
+     * Get metric summary detail filter by summary uuid for Metric Center.
+     *
+     * @param uuid UUID of registered metric summary.
+     * @return A single :py:class:`ai_flow.meta.metric_meta.MetricSummary` object.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java:1073]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java [hunk 10]
```diff
@@ -926,4 +929,167 @@ public void stopListenEvent(String namespace, String key) {
...
+
+    /***
+     * List of metric summaries filter by metric summary fields for Metric Center.
+     *
+     * @param metricName    Name of filtered metric summary.
+     * @param metricKey     Key of filtered metric summary.
+     * @param modelVersion  Version of the model version associated with the registered metric summary.
+     * @param startTime     Start time for timestamp filtered metric summary.
+     * @param endTime       End time for timestamp filtered metric summary.
+     * @return List of :py:class:`ai_flow.meta.metric_meta.MetricSummary` objects.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java:1088]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java [hunk 11]
```diff
@@ -0,0 +1,318 @@
...
+     * @param datasetName   Name of the dataset associated with the registered metric meta.
+     * @param modelName     Name of the model associated with the registered metric meta.
+     * @param jobName       Name of the job associated with the registered metric meta.
+     * @param startTime     Start time of registered metric meta.
+     * @param endTime       End time of registered metric meta.
+     * @param uri           Uri of registered metric meta.
+     * @param tags          Tags of registered metric meta.
+     * @param properties    Properties of registered metric meta.
+     *
+     * @return A single :py:class:`ai_flow.meta.metric_meta.MetricMeta` object.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java:74]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java [hunk 12]
```diff
@@ -0,0 +1,318 @@
...
+    public MetricClient(String target) {
+        this(ManagedChannelBuilder.forTarget(target).usePlaintext().build());
+    }
+
+    public MetricClient(Channel channel) {
+        this.metricServiceStub = MetricServiceGrpc.newBlockingStub(channel);
+    }
+
+    /**
+     * Register a MetricMeta in metric center.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java:59]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java [hunk 13]
```diff
@@ -0,0 +1,318 @@
...
+
+        if (0 == response.getReturnCode()) {
+             return MetricMeta.buildMetricMeta(response.getMetricMeta());
+        } else {
+            throw new AIFlowException(String.valueOf(response.getReturnCode()), response.getReturnMsg());
+        }
+    }
+
+    /**
+     * Update a MetricMeta in metric center.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java:93]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java [hunk 14]
```diff
@@ -0,0 +1,318 @@
...
+     * @param datasetName   Name of the dataset associated with the registered metric meta.
+     * @param modelName     Name of the model associated with the registered metric meta.
+     * @param jobName       Name of the job associated with the registered metric meta.
+     * @param startTime     Start time of registered metric meta.
+     * @param endTime       End time of registered metric meta.
+     * @param uri           Uri of registered metric meta.
+     * @param tags          Tags of registered metric meta.
+     * @param properties    Properties of registered metric meta.
+     *
+     * @return A single :py:class:`ai_flow.meta.metric_meta.MetricMeta` object.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java:107]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java [hunk 15]
```diff
@@ -0,0 +1,318 @@
...
+        } else {
+            return false;
+        }
+    }
+
+    /***
+     * Get metric metadata detail filter by metric name for Metric Center.
+     *
+     * @param metricName Name of registered metric meta. This is expected to be unique in the backend store.
+     * @return A single :py:class:`ai_flow.meta.metric_meta.MetricMeta` object.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java:146]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java [hunk 16]
```diff
@@ -0,0 +1,318 @@
...
+            throw new AIFlowException(String.valueOf(response.getReturnCode()), response.getReturnMsg());
+        }
+    }
+
+    /***
+     * List dataset metric metadata filter by dataset name and project name for Metric Center.
+     *
+     * @param datasetName Name of the dataset associated with the registered metric meta.
+     * @param projectName Name of the project associated with the registered metric meta.
+     * @return List of :py:class:`ai_flow.meta.metric_meta.MetricMeta` objects.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java:164]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java [hunk 17]
```diff
@@ -0,0 +1,318 @@
...
+            throw new AIFlowException(String.valueOf(response.getReturnCode()), response.getReturnMsg());
+        }
+    }
+
+    /***
+     * List model metric metadata filter by model name and project name for Metric Center.
+     *
+     * @param modelName     Name of the model associated with the registered metric meta.
+     * @param projectName   Name of the project associated with the registered metric meta.
+     * @return List of :py:class:`ai_flow.meta.metric_meta.MetricMeta` objects.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java:183]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java [hunk 18]
```diff
@@ -0,0 +1,318 @@
...
+    /***
+     * Register metric summary in Metric Center.
+     *
+     * @param metricName        Name of registered metric summary.
+     * @param metricKey         Key of registered metric summary.
+     * @param metricValue       Value of registered metric summary.
+     * @param metricTimestamp   Timestamp of registered metric summary.
+     * @param modelVersion      Version of the model version associated with the registered metric summary.
+     * @param jobExecutionId    ID of the job execution associated with the registered metric summary.
+     * @return A single :py:class:`ai_flow.meta.metric_meta.MetricSummary` object.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java:206]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java [hunk 19]
```diff
@@ -0,0 +1,318 @@
...
+     * Update metric summary in Metric Center.
+     *
+     * @param uuid              UUID of registered metric summary.
+     * @param metricName        Name of registered metric summary.
+     * @param metricKey         Key of registered metric summary.
+     * @param metricValue       Value of registered metric summary.
+     * @param metricTimestamp   Timestamp of registered metric summary.
+     * @param modelVersion      Version of the model version associated with the registered metric summary.
+     * @param jobExecutionId    ID of the job execution associated with the registered metric summary.
+     * @return A single :py:class:`ai_flow.meta.metric_meta.MetricSummary` object.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java:234]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java [hunk 20]
```diff
@@ -0,0 +1,318 @@
...
+        } else {
+            return false;
+        }
+    }
+
+    /***
+     * Get metric summary detail filter by summary uuid for Metric Center.
+     *
+     * @param uuid UUID of registered metric summary.
+     * @return A single :py:class:`ai_flow.meta.metric_meta.MetricSummary` object.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java:272]: Ditto.

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java [hunk 21]
```diff
@@ -0,0 +1,318 @@
...
+
+    /***
+     * List of metric summaries filter by metric summary fields for Metric Center.
+     *
+     * @param metricName    Name of filtered metric summary.
+     * @param metricKey     Key of filtered metric summary.
+     * @param modelVersion  Version of the model version associated with the registered metric summary.
+     * @param startTime     Start time for timestamp filtered metric summary.
+     * @param endTime       End time for timestamp filtered metric summary.
+     * @return List of :py:class:`ai_flow.meta.metric_meta.MetricSummary` objects.
```
Comment (SteNicholas) [inline flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/MetricClient.java:293]: Ditto.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java (hunk 1 above)
Comment (jiangxin369) [reply flink-ai-flow/ai_flow/java/client/src/main/java/com/aiflow/client/AIFlowClient.java:949]: Corrected

**Review by SteNicholas — APPROVED** [review]

**Pull request merged**